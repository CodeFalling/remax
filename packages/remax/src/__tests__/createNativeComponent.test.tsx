import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import createNativeComponent from '../createNativeComponent';

describe('createNativeComponent', () => {
  it('create native component', () => {
    const Card = createNativeComponent('card');
    const testRenderer = TestRenderer.create(<Card />);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('rename ref', () => {
    const Card = createNativeComponent('card');
    const card = React.createRef();
    const testRenderer = TestRenderer.create(<Card ref={card} />);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
