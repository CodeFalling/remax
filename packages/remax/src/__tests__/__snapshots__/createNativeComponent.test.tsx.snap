// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createNativeComponent create native component 1`] = `<card />`;

exports[`createNativeComponent rename ref 1`] = `
<card
  __ref={[Function]}
/>
`;
